src/pages/Home.vue

```diff
@@ -1,113 +1,77 @@
 <template>
-  <div>
-    <IdEntry
+  <!-- Entry component -->
+  <div v-if="isAuthTypeID && programData">
+    <Entry
       :redirectTo="redirectTo"
       :redirectID="redirectID"
       :purpose="purpose"
       :purposeParams="purposeParams"
-      :textObject="textObject"
-      :inputObject="inputObject"
-      :maxNumberOfInput="maxNumberOfInput"
-      :dataSourceObject="dataSourceObject"
+      :programData="programData"
       :program="program"
       :authType="authType"
     />
   </div>
 </template>
 
 <script>
-import IdEntry from "@/components/IdEntry.vue";
-import firebaseAPI from "@/services/API/getProgramData.js";
+import Entry from "@/components/Entry.vue";
+import programAPIService from "@/services/API/programData.js";
 
 export default {
   name: "Home",
   components: {
-    IdEntry,
+    Entry,
   },
   props: {
+    /** The resource we are redirecting to. Eg. redirectTo = plio tells us that we are redirecting to a plio. */
     redirectTo: {
-      /** The resource we are redirecting to. Eg. redirectTo = plio tells us that we are redirecting to a plio. */
       default: "",
       type: String,
     },
+    /** ID of the resource. Eg. the plioID */
     redirectID: {
-      /** ID of the resource. Eg. the plioID */
       default: "",
       type: String,
     },
+    /** General category of why the data is being captured. Eg: attendance */
     purpose: {
-      /** General category of why the data is being captured. Eg: attendance */
       default: "",
       type: String,
     },
+    /** Subcategory of the purpose. Eg: plio -> means the attendance is for a plio link */
     purposeParams: {
-      /** Subcategory of the purpose. Eg: plio -> means the attendance is for a plio link */
       default: "",
       type: String,
     },
+    /** The program the user falls under. Eg: HaryanaStudents, DelhiStudents */
     program: {
```

Comment:
programName?

Reply:
Decided, no? This will be last step after merging the OTP and youtube PR.